### PR TITLE
Resource related to Kubernetes service, deployment and GCR

### PR DESCRIPTION
**Pull request by savaliyagautam**

### CODE REVIEW

File: name.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+resource "random_string" "resource_suffix" {
+  length  = 5
+  lower   = true
+  upper   = false
+  number  = false
+  special = false
+}
+
+locals {
+  service_name  = var.service_name != "" ? var.service_name : "${data.ns_workspace.this.block_ref}"
```
Comment (BSick7) [inline name.tf]: We should drop the `service_name` var and always use `block_name` (not `block_ref`)

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  default     = "512Mi"
+}
+
+variable "service_cpu" {
+  type        = string
+  description = "Allocated cpu to pods."
+  default     = "0.5"
+}
+
+variable "pods_replica_count" {
```
Comment (BSick7) [inline variables.tf]:
```suggestion
variable "service_count" {
```

This will match the fargate service.

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  type        = number
+  description = "The number of desired pods replicas."
+  default     = 1
+}
+
+variable "container_port" {
```
Comment (BSick7) [inline variables.tf]:
```suggestion
variable "service_port" {
```

Let's stay consistent with fargate service.

**Review by BSick7 — COMMENTED** [review]
A couple small things, but we should be ready to start launching/destroying end-to-end.

File: name.tf (hunk 1 above)
Comment (savaliyagautam) [reply name.tf]: Removed input var `service_name` and replace references with **block_name**

File: variables.tf (hunk 2 above)
Comment (savaliyagautam) [reply variables.tf]: Taken care

File: variables.tf (hunk 3 above)
Comment (savaliyagautam) [reply variables.tf]: Taken care.

**Pull request merged**